src/borg/archiver.py

```diff
@@ -4230,6 +4230,9 @@ def run(self, args):
         args.progress |= is_serve
         self._setup_implied_logging(vars(args))
         self._setup_topic_debugging(args)
+        if all((hasattr(args, x) for x in ('stats', 'dry_run'))) and all((args.stats, args.dry_run)):
```

Comment:
you could combine `hasattr` and the actual access of `args.*` like:
```
if getattr(args, 'stats', False) and getattr(args, 'dry_run', False): ...
```

Reply:
having this code here means that `--stats` is *never* supported when using `--dry-run` (which might be true, did you check all places?).

Reply:
thanks for refactoring.
i checked `archiver.py`,`docs`,`tests`
tests have nothing but the one i changed.
docs have them for create and delete.
archiver.py `do_create` and `do_delete` and i think `--stats` is never supported when using `--dry-run`

Reply:
is there is a possibility for adding a command in the future which runs both `--stats` and `--dry-run` together? if yes, then i think we should add the restriction per command.
maybe define a function  in `archiver`,and call it on `do_create`, `do_prune`, `do_recreate` (and on any new new method where the restriction applies)
```python
def check_stats_and_dry_run(self, command):
    self.print_error("--stats does not work with --dry-run with %s.", command)
    return set_ec(self.exit_code)

```

Reply:
well, that of course depends on the meaning of the options for that command.

for the ones you checked (what we currently have), it does not make sense.

so it is ok for now and in case that will change in future, we need to change the code for that.
but that will be rather obvious, so no problem, keep it as it is (except doing the simplification I suggested).

Reply:
Done.
by the way i edited the comment above to clarify what i meant because i forgot a line in the function.